File: 2951/main.go
package main

import "fmt"

func main() {
	fmt.Println(findPeaks([]int{2, 4, 4}))
	fmt.Println(findPeaks([]int{1, 4, 3, 8, 5}))

}

func findPeaks(mountain []int) []int {
	ans := []int{}
	for i := 1; i < len(mountain) - 1; i++ {
		if mountain[i] > mountain[i-1] && mountain[i] > mountain[i+1] {
			ans = append(ans, i)
		}
	}
	return ans
}

File: 2951/main.rs

struct Solution {}

impl Solution {
    pub fn find_peaks(mountain: Vec<i32>) -> Vec<i32> {
        let mut ans: Vec<i32> = Vec::new();
        for m in 1..mountain.len()-1 {
            if mountain[m] > mountain[m-1] && mountain[m] > mountain[m+1] {
                ans.push(m as i32)
            }
        }
        ans
    }
}

fn main() {
    println!("{:#?}", Solution::find_peaks(vec![1, 4, 3, 8, 5]))
}

